src/vm_cpu.rs

```diff
@@ -327,18 +336,23 @@ impl<'vcpu, 'pages, T: GuestStagePageTable> ActiveVmCpu<'vcpu, 'pages, T> {
                     self.state.trap_csrs.htval << 2 | self.state.trap_csrs.stval & 0x03,
                     self.guest_id,
                 );
-                VmCpuExit::PageFault(
-                    Exception::from_scause_reason(self.state.trap_csrs.scause).unwrap(),
+                VmCpuExit::PageFault {
+                    exception: Exception::from_scause_reason(self.state.trap_csrs.scause).unwrap(),
                     fault_addr,
-                )
+                    fault_pc: RawAddr::guest_virt(self.state.guest_regs.sepc, self.guest_id),
```

Comment:
Is the fault guaranteed to be guest virtual? The HL instructions will handle if it's guest physical and the guest has bare translations set up? Maybe worth a comment.